src/Compatibility/ApiCompat/Microsoft.DotNet.PackageValidation/Package.cs

```diff
@@ -94,11 +94,11 @@ public class Package
             _contentItemCollection = new ContentItemCollection();
             _contentItemCollection.Load(packageAssets);
 
-            PackageAssets = GetContentItemsFromPattern(_conventions.Patterns.AnyTargettedFile);
-            RefAssets = GetContentItemsFromPattern(_conventions.Patterns.CompileRefAssemblies);
-            LibAssets = GetContentItemsFromPattern(_conventions.Patterns.CompileLibAssemblies);
+            PackageAssets = _contentItemCollection.FindItems(_conventions.Patterns.AnyTargettedFile);
```

Comment:
Actually, I'd love to make it obsolete. Originally I reverted it to unblock the insertion.

Please use PopulateItemGroups here.

Reply:
cc @ericstj

Reply:
@nkolev92 when was that method added?  Can we count on it in the version of NuGet we need to reference?

Reply:
Even with the concern from https://github.com/dotnet/sdk/issues/23301 ?

> The newly introduced method PopulateItemGroups does too much (allocating groups that we'd have to flatten).

Reply:
PopulateItemGroups was added in https://github.com/NuGet/NuGet.Client/commit/bcd6856405ebaac80ea63819d0b5a7d87784eae5, it's been available since NuGet 6.1, matching 6.0.200 of the .NET SDK, when https://github.com/dotnet/sdk/issues/23301 originally was raised.

PopulateItemGroups groups them by framework, by what would be chosen. 
Do you think that'd bring anything in your validation? 
I already see there's some lookups for frameworks supported, but didn't look at the code in depth.

Reply:
Thanks for the reminder, @akoeplinger.  What's the problem with `FindItems` -- I thought @jebriede mentioned he fixed it, and it looks like switching to use `PopulateItemGroups` requires that we allocate to get the flat list of items.

Reply:
NuGet doesn't use the API (looking at the 6.1 changes, it wasn't used before that either).
The packages we ship did not go through a thorough public API review (it was inherited) and trying to preserve compat in all these public methods has affected when trying to do perf work.

Reply:
Just to clarify things, it's not marked as obsolete anymore, so you can continue using it if it's really the only way for you.
I misunderstood the way you were using the API, since NuGet side we always care about the groups.

Example:

lib/net5.0/a.dll
lib/net5.0/b.dll
lib/net6.0/a.dll 

We care about the lib/net5.0 and lib/net6.0 groups since it's one or the other, but your validation might not care about that. 

If you think the groups would be valuable, you can use that method and I can mark it as obsolete again.
Otherwise, you can just do this revert and we won't make changes NuGet side. 

The only gain of marking that method as obsolete for us is simpler/smaller API.

Reply:
There's also this usage in source-build-reference-packages: https://github.com/dotnet/source-build-reference-packages/blob/0d066e61a30c2599d0ced871ea45acf0e10571af/src/packageSourceGenerator/PackageSourceGeneratorTask/GetPackageItems.cs#L110

I think if it'd make your lifes easier for the perf work we could move to `PopulateItemGroups` like we did here but if it's just a nice to have cleanup right now I'd prefer keeping the existing `FindItems` code.

Reply:
You can just keep it. 
My hope is that I made the current codepath efficient enough that it's a long time until someone looks at that again :D

Reply:
ok :) can someone please approve this PR then?